examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -156,11 +158,21 @@ function implicit_tendency!(Yₜ, Y, p, t)
 end
 
 function remaining_tendency!(Yₜ, Y, p, t)
+    (; is_distributed) = p
     Yₜ .= zero(eltype(Yₜ))
     default_remaining_tendency!(Yₜ, Y, p, t)
     additional_tendency!(Yₜ, Y, p, t)
-    Spaces.weighted_dss!(Yₜ.c, p.ghost_buffer.c)
-    Spaces.weighted_dss!(Yₜ.f, p.ghost_buffer.f)
+    if is_distributed
+        Spaces.weighted_dss_init_comms!(Yₜ.c, p.ghost_buffer.c)
+        Spaces.weighted_dss_init_comms!(Yₜ.f, p.ghost_buffer.f)
+        Spaces.weighted_dss_internal!(Yₜ.c, p.ghost_buffer.c)
+        Spaces.weighted_dss_internal!(Yₜ.f, p.ghost_buffer.f)
+        Spaces.weighted_dss_ghost!(Yₜ.c, p.ghost_buffer.c)
+        Spaces.weighted_dss_ghost!(Yₜ.f, p.ghost_buffer.f)
+    else
+        Spaces.weighted_dss!(Yₜ.c, p.ghost_buffer.c)
+        Spaces.weighted_dss!(Yₜ.f, p.ghost_buffer.f)
+    end
```

Comment:
I think it would be simpler to define `weighted_dss_init_comms!` and `weighted_dss_ghost!` as no-ops (e.g. `return nothing`) for non-distributed topologies, then you don't need the branch.